### PR TITLE
Timeago

### PR DESCRIPTION
**Pull request by keithamus**
### What?

This adds a new `TimeAgo` component based on the `TimeAgoComponent` from dotcom. Small changes have been made to get it to comply with the Primer tests but its 99.9% the same component.

### Why?

This is a useful component to have in the Primer namespace, and should live in Primer ViewComponents where it can better serve users of PVC.

This is also one of the first behavioural components which will help validate JS behaviours in PVC. Similar to https://github.com/primer/view_components/pull/282

### CODE REVIEW

File: test/components/time_ago_component_test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+componentTest('time_ago_component', it => {
+
+  it('works', async page => {
+    await expect(page).to.have.selector('body')
+    const el = await page.$('body')
+    await expect(el).to.be.visible()
+  })
+
+})
```
Comment (manuelpuyol) [inline test/components/time_ago_component_test.js:12]: does this file do something? 
I couldn't find `./helpers.js` and don't see `chai` as a dependency

Comment (keithamus) [reply test/components/time_ago_component_test.js:12]: Oops this was from an old iteration of the tests!

File: app/components/time_ago_component.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Primer
+  # A basic wrapper for the "<time-ago>" element.
+  class TimeAgoComponent < Primer::Component
+    def initialize(time:, micro: false, **system_arguments)
```
Comment (manuelpuyol) [inline app/components/time_ago_component.rb:6]: mind adding some docs here?
e.g: https://github.com/primer/view_components/blob/776d4ecee07f102ef16e41b3245441449d0963db/app/components/primer/avatar_component.rb#L10-L24

File: app/components/time_ago_component.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Primer
+  # A basic wrapper for the "<time-ago>" element.
```
Comment (manuelpuyol) [inline app/components/time_ago_component.rb:4]: I think we need something more descriptive here. we normally follow something like `Use TimeAgo to ...`

File: app/components/time_ago_component.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  class TimeAgoComponent < Primer::Component
+    def initialize(time:, micro: false, **system_arguments)
+      @system_arguments = system_arguments
+      @system_arguments[:datetime] = time.utc.iso8601
+      @system_arguments[:classes] = class_names("no-wrap", @system_arguments[:classes])
+      @system_arguments[:tag] = "time-ago"
+      @system_arguments[:format] = "micro" if micro
+      @content = if micro
```
Comment (manuelpuyol) [inline app/components/time_ago_component.rb:12]: nit: might we call this `@time`?
`@content` is an internal ivar from the view_component framework

File: app/components/time_ago_component.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Primer
+  # A basic wrapper for the "<time-ago>" element.
+  class TimeAgoComponent < Primer::Component
+    def initialize(time:, micro: false, **system_arguments)
+      @system_arguments = system_arguments
+      @system_arguments[:datetime] = time.utc.iso8601
+      @system_arguments[:classes] = class_names("no-wrap", @system_arguments[:classes])
```
Comment (manuelpuyol) [inline app/components/time_ago_component.rb:9]: do we **always** want `no-wrap`?

File: app/components/time_ago_component.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+import {TimeAgoElement} from '@github/time-elements'
```
Comment (manuelpuyol) [inline app/components/time_ago_component.ts:1]: for the future, shall we setup eslint for this project?

File: app/components/time_ago_component.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+module Primer
+  # Use Primer::TimeAgoComponent to display a time relative to how long ago it was
+  class TimeAgoComponent < Primer::Component
+
+    #
+    # @example auto|Default
+    #   <%= render(Primer::TimeAgoComponent.new(time: Time.zone.now)) %>
+    #
+    # @param time [Time] The time to be formatted 
+    # @param micro [Boolean] If true then the text will be formatted in "micro" mode, using as few characters as possible
```
Comment (manuelpuyol) [inline app/components/time_ago_component.rb:12]:
```suggestion
    # @param micro [Boolean] If true then the text will be formatted in "micro" mode, using as few characters as possible
    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
```

File: app/components/time_ago_component.ts (hunk 6 above)
Comment (keithamus) [reply app/components/time_ago_component.ts:1]: eslint would be a fantastic idea!

File: app/components/time_ago_component.rb (hunk 5 above)
Comment (keithamus) [reply app/components/time_ago_component.rb:17]: We had this in the helper/component on GitHub. I think we do. I can't think of a good reason to remove it.

File: app/components/time_ago_component.rb (hunk 4 above)
Comment (keithamus) [reply app/components/time_ago_component.rb]: I've refactored this out more so the initialiser just assigns variables an does not apply logic, the logic is then in a `time_in_words` method.

File: app/components/time_ago_component.rb [hunk 8]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use Primer::TimeAgoComponent to display a time relative to how long ago it was
```
Comment (srt32) [inline app/components/time_ago_component.rb:4]:
```suggestion
  # Use Primer::TimeAgoComponent to display a time relative to how long ago it was. This component requires JavaScript`.
```

File: static/statuses.json [hunk 9]
```diff
@@ -1 +1,5 @@
...
+>>>>>>> origin/main
```
Comment (srt32) [inline static/statuses.json:5]: Your next commit should force the docs action to fix this for you.

File: CHANGELOG.md [hunk 10]
```diff
@@ -55,6 +55,10 @@
 
     *Simon Taranto*
 
+* Add `TimeAgo` component.
```
Comment (manuelpuyol) [inline CHANGELOG.md:58]: may we move this to be below `# main`?

File: app/components/primer/time_ago_component.ts [hunk 11]
```diff
@@ -0,0 +1 @@
+import '@github/time-elements'
```
Comment (manuelpuyol) [inline app/components/primer/time_ago_component.ts:1]: shouldn't we only import the time-ago element? `import {TimeAgoElement} from '@github/time-elements'`

File: app/components/primer/time_ago_component.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use Primer::TimeAgoComponent to display a time relative to how long ago it was. This component requires JavaScript.
```
Comment (joelhawksley) [inline app/components/primer/time_ago_component.rb:4]: Might we want to link to some sort of documentation on how to load the JS?

**Review by joelhawksley — APPROVED** [review]
👏🏻

File: app/components/primer/time_ago_component.ts (hunk 11 above)
Comment (keithamus) [reply app/components/primer/time_ago_component.ts:1]: We should but:
 
 - Our linter complains
 - Or compiler doesn't actually tree shake this as the whole file has side effects meaning no dead code elimination can take place

This is a piece of tech debt we should address upstream and patch in a follow up PR I think.

File: app/components/primer/time_ago_component.rb (hunk 12 above)
Comment (keithamus) [reply app/components/primer/time_ago_component.rb:4]: Let's tackle this in a follow up PR for all elements requiring JS

**Review by srt32 — APPROVED** [review]

**Pull request merged**